community/kernel/src/main/java/org/neo4j/kernel/impl/proc/ReflectiveProcedureCompiler.java

```diff
@@ -588,6 +588,12 @@ public ProcedureSignature signature()
                             numberOfDeclaredArguments, input.length );
                 }
 
+                // Some input fields are not supported by Cypher and need to be mapped
+                for ( int i = 0; i < input.length; i++ )
+                {
+                    input[i] = signature.inputSignature().get( i ).map( input[i] );
```

Comment:
`signature.inputSignature` should probably be assigned to a local variable outside the loop

I find it a bit scary to modify `input` in place here, are we sure the caller are (or will in the future) use the array for other things where this mapping is not expected?

Reply:
I have written an extra test to verify that the values passed in are not modified. I see your concern that this is not safe against future usages of this system, but I thought that should be caught by tests. I'm hoping to minimize object creation, which will happen on every single call if we do not do this. Now we only take a cost on byte[] processing, not all other types.

Reply:
I just a week ago we did do an extra copy here so I am not that worried about the performance.